pkg/pullsync/pullstorage/pullstorage_test.go

```diff
@@ -0,0 +1,376 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pullstorage_test
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/localstore"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/pullsync/pullstorage"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/storage/mock"
+	stesting "github.com/ethersphere/bee/pkg/storage/testing"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var (
+	addrs = []swarm.Address{
+		swarm.MustParseHexAddress("0001"),
+		swarm.MustParseHexAddress("0002"),
+		swarm.MustParseHexAddress("0003"),
+		swarm.MustParseHexAddress("0004"),
+		swarm.MustParseHexAddress("0005"),
+		swarm.MustParseHexAddress("0006"),
+	}
+	limit = 5
```

Comment:
the limit here at the tests is injected through the function call. however, pull sync protocol enforces its own limit which is different (right now set at 50), some of the test cases around the pull sync protocol are also checking around the limit (of 50 - `pullsync.go#L37`) and so this is a bit of a debt that needs to be settled. up to you guys if this should be during this PR or not